backend/Alpaki.Logic/Handlers/PasswordRecovery/ForgotPassword/ForgotPasswordHandler.cs

```diff
@@ -0,0 +1,94 @@
+    {
+        private readonly IDatabaseContext _dbContext;
+        private readonly IMailService _mailService;
+        private readonly ISystemClock _systemClock;
+        private readonly LinkGenerator _linkGenerator;
+        private readonly IHttpContextAccessor _contextAccessor;
+        private readonly MailKitOptions _options;
+
+        public ForgotPasswordHandler(IDatabaseContext dbContext, 
+            IMailService mailService, 
+            IOptions<MailKitOptions> options,
+            ISystemClock systemClock,
+            LinkGenerator linkGenerator,
+            IHttpContextAccessor contextAccessor)
+        {
+            _dbContext = dbContext;
+            _mailService = mailService;
+            _systemClock = systemClock;
+            _linkGenerator = linkGenerator;
+            _contextAccessor = contextAccessor;
+            _options = options.Value;
+        }
+        public async Task<ForgotPasswordResponse> Handle(ForgotPasswordRequest request, CancellationToken cancellationToken)
+        {
+            var user = await _dbContext.Users.SingleOrDefaultAsync(x => x.Email == request.Email, cancellationToken: cancellationToken);
+            if (user == null)
+            {
+                return new ForgotPasswordResponse();
+            }
+
+            var token = GetToken(user);
+
+            var link = GenerateLink(user, token);
+            
+            var message = MessageBuilder
+                .Create()
+                .WithReceiver(user.Email)
+                .WithSender(_options.Email)
+                .WithSubject(MessageTemplates.PasswordReset.Subject)
+                .WithBody(MessageTemplates.PasswordReset.Body, link)
+                .Build();
+
+            await _mailService.Send(message);
+
+            return new ForgotPasswordResponse();
+        }
+
+        private string GetToken(User user)
+        {
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var key = Encoding.ASCII.GetBytes(user.PasswordHash);
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(new[]
+                {
+                    new Claim(ClaimTypes.NameIdentifier, user.UserId.ToString()),
+                    new Claim(ClaimTypes.Name, user.UserId.ToString()),
+                    new Claim(ClaimTypes.Role, ((int)user.Role).ToString()), 
+                }),
+                Expires = _systemClock.UtcNow.AddHours(1).UtcDateTime,
+                SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key), SecurityAlgorithms.HmacSha256Signature)
+            };
+            var token = tokenHandler.CreateToken(tokenDescriptor);
+            return tokenHandler.WriteToken(token);
+        }
+
+        private string GenerateLink(IUser user, string token)
+        {
+            return _linkGenerator.GetUriByPage(_contextAccessor.HttpContext, "/PasswordReset",null, new {userId = user.UserId, token=token});
```

Comment:
nope. nie możemy generować linku. my przecież robimy api konsumowane prze aplikację i to ona zna link. przecież to może być jakiś meta link otwierający app-store albo coś takiego. na pewno nie chcesz żeby user wylądował na twoim endpoincie api

Reply:
No tak, ale co innego wysłać w mailu, żeby użytkownik mógł jakoś zmienić to hasło? Można kod wysyłać, tylko to będzie mało wygodne dla użytkownika, bo kod będzie długi, żeby go nie można było zbrutforsować.

Reply:
podobny kod jak przy zaproszeniach może np 5-6 znaków. zapisany w bazie przypisany do usera z czasem ważnośći 24h